features/issues/api/use-get-issues.tsx

```diff
@@ -2,32 +2,39 @@ import { useEffect } from "react";
 import { useQuery, useQueryClient } from "@tanstack/react-query";
 import { getIssues } from "@api/issues";
 import type { Page } from "@typings/page.types";
-import type { Issue } from "@api/issues.types";
-
+import type { Issue, QueryFilters } from "@api/issues.types";
+import { useFilters } from "../components/filters/use-filters";
 const QUERY_KEY = "issues";
 
-export function getQueryKey(page?: number) {
+export function getQueryKey(page?: number, filters?: QueryFilters) {
   if (page === undefined) {
     return [QUERY_KEY];
   }
-  return [QUERY_KEY, page];
+  return [QUERY_KEY, page, filters];
 }
 
 export function useGetIssues(page: number) {
+  const { filters } = useFilters();
   const query = useQuery<Page<Issue>, Error>(
     getQueryKey(page),
-    ({ signal }) => getIssues(page, { signal }),
+    ({ signal }) => getIssues(page, filters, { signal }),
     { keepPreviousData: true }
   );
 
   // Prefetch the next page!
   const queryClient = useQueryClient();
   useEffect(() => {
     if (query.data?.meta.hasNextPage) {
-      queryClient.prefetchQuery(getQueryKey(page + 1), ({ signal }) =>
-        getIssues(page + 1, { signal })
+      queryClient.prefetchQuery(getQueryKey(page + 1, filters), ({ signal }) =>
+        getIssues(page + 1, filters, { signal })
       );
     }
-  }, [query.data, page, queryClient]);
+  }, [query.data, page, filters, queryClient]);
```

Comment:
Not sure if this is creating your problem but the filters object gets created during each render. So adding it as a dependency to this useEffect causes it to get executed on each render. You could add each filter value as a separate dependency. Or use '...Object.values(filter)' as dependency which should have the same effect (if the values are always returned in the same order).

The same might be valid for the query key but I'm not sure about that. You might want to check the react-query docs about using objects as query keys